perfkitbenchmarker/linux_benchmarks/specsfs2014_benchmark.py

```diff
@@ -0,0 +1,264 @@
+
+SPEC SFS 2014 homepage: http://www.spec.org/sfs2014/
+
+In order to run this benchmark copy your 'SPECsfs2014_SP1.iso'
+and 'netmist_license_key' files into the data/ directory.
+"""
+
+import posixpath
+import xml.etree.ElementTree
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import data
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker import sample
+
+from perfkitbenchmarker.linux_packages import gluster
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    'specsfs2014_config', None,
+    'This flag can be used to specify an alternate SPEC config file to use. '
+    'If this option is specified, none of the other benchmark specific flags '
+    'which operate on the config file will be used (since the default config '
+    'file will be replaced by this one).')
+flags.DEFINE_enum(
+    'specsfs2014_benchmark', 'VDI', ['VDI', 'DATABASE', 'SWBUILD', 'VDA'],
+    'The SPEC SFS 2014 benchmark to run.')
+flags.DEFINE_integer(
+    'specsfs2014_load', 1,
+    'The starting load in units of SPEC "business metrics". This varies '
+    'from benchmark to benchmark.',
+    lower_bound=1)
+flags.DEFINE_integer(
+    'specsfs2014_incr_load', 1,
+    'The amount to increment "load" by for each run.',
+    lower_bound=1)
+flags.DEFINE_integer(
+    'specsfs2014_num_runs', 1,
+    'The total number of SPEC runs. The load for the nth run is '
+    '"load" + n * "specsfs_incr_load".',
+    lower_bound=1)
+
+BENCHMARK_NAME = 'specsfs2014'
+BENCHMARK_CONFIG = """
+specsfs2014:
+  description: >
+    Run SPEC SFS 2014. For a full explanation of all benchmark modes
+    see http://www.spec.org/sfs2014/.
+  vm_groups:
+    clients:
+      vm_spec: *default_single_core
+      vm_count: null
+    gluster_servers:
+      vm_spec: *default_single_core
+      disk_spec: *default_500_gb
+      vm_count: 3
+"""
+
+SPEC_SFS_2014_ISO = 'SPECsfs2014_SP1.iso'
+SPEC_SFS_2014_LICENSE = 'netmist_license_key'
+SPEC_DIR = 'spec'
+
+VOLUME_NAME = 'gv0'
+MOUNT_POINT = '/scratch'
+PUBLISHED_METRICS = frozenset([
+    'achieved rate', 'average latency', 'overall throughput',
+    'read throughput', 'write throughput'
+])
+METADATA_KEYS = frozenset([
+    'op rate', 'run time', 'processes per client', 'file size',
+    'client data set size', 'starting data set size', 'initial file space',
+    'maximum file space'
+])
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  data.ResourcePath(SPEC_SFS_2014_ISO)
+  data.ResourcePath(SPEC_SFS_2014_LICENSE)
+  if FLAGS.specsfs2014_config:
+    data.ResourcePath(FLAGS.specsfs2014_config)
+
+
+def _PrepareClient(vm, gluster_server):
+  """Prepares a SPEC client by copying SPEC to the VM and mounting Gluster."""
+  mount_dir = 'spec_mnt/'
```

Comment:
For directories created by the benchmark's Prepare stage, consider an error if they already exist, and remove them in Cleanup.